doc/source/whatsnew/v0.19.0.txt

```diff
@@ -855,3 +855,4 @@ Bug Fixes
 
 - Bug in ``.to_excel()`` when DataFrame contains a MultiIndex which contains a label with a NaN value (:issue:`13511`)
 - Bug in ``pd.read_csv`` in Python 2.x with non-UTF8 encoded, multi-character separated data (:issue:`3404`)
+- Bug in ``pd.read_hdf`` may raise ``ValueError`` if the mode supplied is ``w`` (:issue:`13623`)
```

Comment:
Currently it raises ``KeyError``, thus description should something like:

-  Bug in ``pd.read_hdf`` may raise ``KeyError`` if the mode supplied is ``w``, rather than ``ValueError``